src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java

```diff
@@ -150,13 +152,22 @@ public Payment getPlatformPayment(String name) {
                 .map(platform1 -> platform1.publishTask(experiment))
                 .orElseThrow(() -> new IllegalArgumentException("Platform not found!"))
                 .handle((s1, throwable) -> {
-                    if (s1 != null) {
+                    //if the creation was successful update the task
+                    if (s1 != null && throwable == null) {
                         result.setPlatformData(s1);
-                    } else {
-                        result.setStatus(TaskStatus.stopped);
+                        if (!tasksOps.updateTask(result)) {
+                            throw new IllegalStateException("Updating record for published task failed");
+                        }
+                    }
+                    //if there is no useful key throw!
+                    if (s1 == null || s1.isEmpty()) {
```

Comment:
Mhmm then just the s1 != null is not needed, s1 could be empty which would result in a for us error case.